common/djangoapps/student/models.py

```diff
@@ -1793,6 +1793,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user).exists():
+                # if user has not profile then create it.
+                user_profile = UserProfile(name=user.username, user=user)
+                user_profile.save()
+
```

Comment:
> Should this UserProfile creation be outside the "PREVENT_CONCURRENT_LOGINS is True" logic?

@doctoryes if ``PREVENT_CONCURRENT_LOGINS=True`` and user try to login then this receiver get singal and handels session to avoid CONCURRENT, it is called before he can actually see landing page. We need user profile to be created before L#1802.
after get pass from this receiver you can see we are creating [user profile here](https://github.com/edx/edx-platform/blob/master/common/djangoapps/external_auth/views.py#L481)

> Also, these lines should be replaced with:
``user_profile, __ = UserProfile.objects.get_or_create(name=user.username, user=user)``

@doctoryes this name param is user full name which is not constant many users can have same name, and user can also edit his name once he login (from profile)